util/reggen/gen_rust.py

```diff
@@ -0,0 +1,394 @@
+    define_name = regname + '_' + as_define(field.name + "_FIELD_WIDTH")
+    define = gen_const(define_name, str(field_width), existing_defines)
+    genout(outstr, define)
+
+    define_name = regname + '_' + as_define(field.name + "_FIELDS_PER_REG")
+    define = gen_const(define_name, str(fields_per_reg), existing_defines)
+    genout(outstr, define)
+
+    define_name = regname + "_MULTIREG_COUNT"
+    define = gen_const(define_name, str(subreg_num), existing_defines)
+    genout(outstr, define)
+
+    genout(outstr, '\n')
+
+
+def gen_const_multireg(outstr: TextIO,
+                         multireg: MultiRegister,
+                         component: str,
+                         regwidth: int,
+                         rnames: Set[str],
+                         existing_defines: Set[str]) -> None:
+    comment = multireg.reg.desc + " (common parameters)"
+    genout(outstr, format_comment(first_line(comment)))
+    if len(multireg.reg.fields) == 1:
+        regname = as_define(component + '_' + multireg.reg.name)
+        gen_multireg_field_defines(outstr, regname, multireg.reg.fields[0],
+                                   len(multireg.regs), regwidth, existing_defines)
+    else:
+        log.warn("Non-homogeneous multireg " + multireg.reg.name +
+                 " skip multireg specific data generation.")
+
+    for subreg in multireg.regs:
+        gen_const_register(outstr, subreg, component, regwidth, rnames,
+                             existing_defines)
+
+
+def gen_cdefines_interrupt_field(outstr: TextIO,
+                                 interrupt: Signal,
+                                 component: str,
+                                 regwidth: int,
+                                 existing_defines: Set[str]) -> None:
+    fieldlsb = interrupt.bits.lsb
+    iname = interrupt.name
+    defname = as_define(component + '_INTR_COMMON_' + iname)
+
+    if interrupt.bits.width() == 1:
+        # single bit
+        genout(
+            outstr,
+            gen_const(defname + '_BIT', str(fieldlsb), existing_defines))
+    else:
+        # multiple bits (unless it is the whole register)
+        if interrupt.bits.width() != regwidth:
+            mask = interrupt.bits.msb >> fieldlsb
+            genout(
+                outstr,
+                gen_const(defname + '_MASK', hex(mask), existing_defines))
+            genout(
+                outstr,
+                gen_const(defname + '_OFFSET', str(fieldlsb),
+                           existing_defines))
+
+
+def gen_const_interrupts(outstr: TextIO,
+                            block: IpBlock,
+                            component: str,
+                            regwidth: int,
+                            existing_defines: Set[str]) -> None:
+    # If no_auto_intr_regs is true, then we do not generate common defines,
+    # because the bit offsets for a particular interrupt may differ between
+    # the interrupt enable/state/test registers.
+    if block.no_auto_intr:
```

Comment:
Nit: The name in the code doesn't match the comment here.

Reply:
Done.